src/builtin-addons/core/script-definition-provider.ts

```diff
@@ -11,16 +14,41 @@ import {
   isServiceInjection,
   isNamedServiceInjection,
   isTemplateElement,
+  isImportSpecifier,
 } from './../../utils/ast-helpers';
 import { normalizeServiceName } from '../../utils/normalizers';
 import { isModuleUnificationApp, podModulePrefixForRoot } from './../../utils/layout-helpers';
 import { provideRouteDefinition } from './template-definition-provider';
+
 type ItemType = 'Model' | 'Transform' | 'Service';
 
 // barking on 'LayoutCollectorFn' is defined but never used  @typescript-eslint/no-unused-vars
 // eslint-disable-line
 type LayoutCollectorFn = (root: string, itemName: string, podModulePrefix?: string) => string[];
 
+const mFindByGlob = memoize(findByGlob, {
+  length: 3,
+  maxAge: 600000,
+});
+
+function findByGlob(pathName: string, appName: string, parentRoot: string) {
+  let res: string[] = [];
+  const pathParts = pathName.split('/');
+  const addonName = pathParts.shift();
+
+  res = fg.sync([`${parentRoot}/${appName}/(lib|engines)/**/${addonName}/**/${pathParts.join('/')}?(/index).js`], { ignore: ['**/node_modules/**'] });
+
+  if (!res.length) {
```

Comment:
This logic should be already covered in packages resolver (including engines and in-repo addons)

If it’s not working for some cases, we should add test to catch issue.

Reply:
https://github.com/lifeart/ember-language-server/blob/component-context-info-origin/src/utils/layout-helpers.ts#L133

Reply:
can we somehow statically resolve this case? For example, reading libs location from package json...

and for engine case I think we could extend path matcher to get more meta over it.

Reply:
ya I think we can do that. The issue for us was mostly the way our app is structured, where one app has sub ember-apps within it, and hence it was freaking out.
The changes that considers parent root as well as app sub roots, should suffice.
I had this in place to be a fallback for any cases that doesnt get covered for any reason. I am fine removing it since I just tested, almost all of our usecases works out well.